### ISSUE TITLE
Using the element "deprecated" is deprecated

### ISSUE THREAD

**Issue by BlackWiCKED** [doctrine/DoctrineBundle#1067]
Got these 3 deprecation warnings recently:

`Using the element "deprecated" is deprecated for the service "Symfony\Bridge\Doctrine\RegistryInterface" which is defined as an alias in ".../vendor/doctrine/doctrine-bundle/DependencyInjection/../Resources/config/dbal.xml". The XmlFileLoader will raise an exception in Symfony 4.0, instead of silently ignoring unsupported elements.`

`Using the element "deprecated" is deprecated for the service "Doctrine\Bundle\DoctrineBundle\Registry" which is defined as an alias in ".../vendor/doctrine/doctrine-bundle/DependencyInjection/../Resources/config/dbal.xml". The XmlFileLoader will raise an exception in Symfony 4.0, instead of silently ignoring unsupported elements.`

`Using the element "deprecated" is deprecated for the service "Doctrine\Common\Persistence\ObjectManager" which is defined as an alias in ".../vendor/doctrine/doctrine-bundle/DependencyInjection/../Resources/config/orm.xml". The XmlFileLoader will raise an exception in Symfony 4.0, instead of silently ignoring unsupported elements.`

I'm using Symfony 3.4.35.

**Comment by BlackWiCKED** [comment]
Possible solution: I removed the affected lines from the config files. Not sure how they got there, but the suggested replacements were already added so it felt a safe thing to do - and it worked well. Maybe it's just project specific thing, in that case ignore my report.

These were the lines I had to remove:

1. /Resources/config/orm.xml
```
<service id="Doctrine\Common\Persistence\ObjectManager" alias="doctrine.orm.entity_manager" public="false">
    <deprecated>The "%alias_id%" service alias is deprecated, use `Doctrine\ORM\EntityManagerInterface` instead.</deprecated>
</service>
```

2. /Resources/config/dbal.xml
```
<service id="Symfony\Bridge\Doctrine\RegistryInterface" alias="doctrine" public="false">
    <deprecated>The "%alias_id%" service alias is deprecated, use `Doctrine\Common\Persistence\ManagerRegistry` instead.</deprecated>
</service>
<service id="Doctrine\Bundle\DoctrineBundle\Registry" alias="doctrine" public="false">
    <deprecated>The "%alias_id%" service alias is deprecated, use `Doctrine\Common\Persistence\ManagerRegistry` instead.</deprecated>
</service>
```

These lines are not in the official repository, so I'm a bit confused how they got there.

**Comment by maxhelias** [comment]
This service is deprecated and remove on doctrine-bundle 2.0. You have this message because Symfony DependencyInjection not support the deprecated feature on 3.4

**Comment by alcaeus** [comment]
@blackWICKED please don’t modify vendor code. Instead, just ignore the deprecation as it is triggered by vendor code anyways. You can ignore it entirely in this case. Instead, whenever you’re using one of these services, stop using them.

When you upgrade to Symfony 4, the deprecation about the “deprecated” Attribute will be gone and you’ll instead receive a deprecation notice when you use one of the deprecated services.

Closing as there is no issue to be fixed here.

**Comment by stof** [comment]
Well, we should avoid making our code throw deprecations though. A solution would be to alter definitions after the fact (in the DI extension) to mark them as deprecated only when Symfony supports it.

**Comment by alcaeus** [comment]
@stof that's a good point, thank you.

@maxhelias want to give this a shot since you took care of the original PR?

**Comment by maxhelias** [comment]
@alcaeus, yes I can take care of that, if ever the https://github.com/symfony/symfony/pull/34587 is not supported in Symfony

**Comment by alcaeus** [comment]
Good point. Let's wait until that is resolved. If that PR is not merged, we'll fix this here. Thanks for taking care of this! ❤️

**Comment by stof** [comment]
well, my own vote on the symfony issue is that it should be rejected, given that it tries to allow using future features in 3.4, which does not make sense (marking aliases as deprecated is a Symfony 4.4+ feature).

**Comment by nicolas-grekas** [comment]
FYI symfony/symfony#34587 has been rejected.

**Comment by alcaeus** [comment]
Thanks! @maxhelias care to open a PR to fix this in 1.12.2?
